### PR TITLE
Fix save conversation exception

### PR DESCRIPTION
**Pull request by ben181231**
connect oursky/skygear-support#53

### CODE REVIEW

File: SKYKitChat/Classes/Core/Records/SKYConversation.m [hunk 1]
```diff
@@ -54,6 +54,32 @@ + (instancetype)recordWithRecord:(SKYRecord *)record
     return conversation;
 }
 
++ (NSArray<NSString *> *)restrictedFieldKeys
+{
+    static NSArray<NSString *> *_restrictedFieldKeys;
+    if (!_restrictedFieldKeys) {
+        // this list is retrieved from:
+        //   https://github.com/SkygearIO/chat/blob/1.6.0-3/chat/conversation_handlers.py#L211
```
Comment (carmenlau) [inline SKYKitChat/Classes/Core/Records/SKYConversation.m:62]: From the plugin code, the record before_save will remove those attributes. So we should not need to do it in the client?

Comment (ben181231) [reply SKYKitChat/Classes/Core/Records/SKYConversation.m:62]: SDK will include serialized message record in `last_message` field and skygear server will expect a record reference. The tricky point is that the schema conflict error occurs before calling `before_save`. 😅

Comment (carmenlau) [reply SKYKitChat/Classes/Core/Records/SKYConversation.m:62]: Oh, than I think we should also check the other SDK if it has similar issue? Let me check and merge this first.

**Comment by carmenlau** [comment]
Merged via a863d25

**Pull request closed without merging**